Ex3/Ex3.java

```diff
@@ -0,0 +1,128 @@
+import java.util.TreeSet;
+
+public class Ex3 {
+	
+	private int rows;
+	private int columns;
+	private char[][] matrix;
+	private Dictionary dictionary;
+	
+	//a set for final valid words in the dictionary that were found in the grid
+	private TreeSet<String> wordsInMatrix = new TreeSet<>();
+	
+	/**
+	 * Constructor
+	 * @param rows number of rows
+	 * @param columns number of columns
+	 * @param matrix Matrix of rows x columns filled with characters of type char
+	 * @param dictionary a Dictionary object
+	 */
+	public Ex3(int rows, int columns, char[][] matrix, Dictionary dictionary) {
+		this.rows = rows;
+		this.columns = columns;
+		this.matrix = matrix;
+		this.dictionary = dictionary;
+
+		wordsInMatrix = findWords(rows, columns, matrix, dictionary);
+	}
+
+	/**
+	 * A method that iterates over the characters in a matrix and
+	 * initiate on each cell the check-up
+	 * @param rows
```

Comment:
FYI: In a Google code review this would get rejected because Javadoc which adds no value we just delete.